src/parcsr_ls/par_mgr_setup.c

```diff
@@ -1108,19 +1119,14 @@ hypre_MGRSetup( void               *mgr_vdata,
          interp_type[lev] = 2;
       }
 
-      /* Extract A_FF and A_FC when needed by MGR's interpolation/relaxation strategies */
-#if !defined(HYPRE_USING_GPU)
-      if ((Frelax_type[lev] == 2)   ||
-          (Frelax_type[lev] == 9)   ||
-          (Frelax_type[lev] == 99)  ||
-          (Frelax_type[lev] == 199) ||
-          (interp_type[lev] == 12)  ||
-          (mgr_coarse_grid_method[lev] != 0))
-#endif
-      {
-         hypre_ParCSRMatrixGenerateFFFC(A_array[lev], CF_marker, coarse_pnts_global,
-                                        NULL, &A_FC, &A_FF);
-      }
+      /* Compute C/F splitting (needed by RAP computation and other operations) */
```

Comment:
It looks like we are doing to the work of generating A_CF even if it might not be needed?

Reply:
you may give NULL pointers to what you don't need. need to double check, not 100% sure.

Reply:
It's not possible:

https://github.com/hypre-space/hypre/blob/6fc944427268ad18b4e3cd29ab2fc7f1b0854658/src/parcsr_mv/gen_fffc.c#L445

https://github.com/hypre-space/hypre/blob/6fc944427268ad18b4e3cd29ab2fc7f1b0854658/src/parcsr_mv/gen_fffc.c#L484

Reply:
OK. The device version can do this.